mlir/lib/Dialect/Tosa/Transforms/TosaFolders.cpp

```diff
@@ -1716,29 +1734,43 @@ struct ReduceConstantOptimization : public OpRewritePattern<OperationType> {
     const auto shapedOldElementsValues =
         denseElementsAttr.getType().cast<ShapedType>();
 
-    if (!llvm::isa<IntegerType>(shapedOldElementsValues.getElementType()))
-      return rewriter.notifyMatchFailure(
-          op, "reduce input currently supported with integer type");
+    if constexpr (FPSupport == false &&
```

Comment:
do you really need it here?

Reply:
perhaps we can only keep `hasFPSupport` and remove the `is_same_v` part?

Reply:
I think so, without 'FPSupport == false', it means the statement "`reduce_any` and `reduce_all` don't support floating point" is the source of truth. But actually the spec doesn't forbid input element type of `reduce_any` and `reduce_all` to be floating point, nor does the verifier check on that. We 'think' it makes no sense to have floating point as input element type because we use logical operators for APFloat as implementation.